File: internal/model/purchase.go
package model

import (
	"time"

	"ampmanager/internal/precision"
)

type PaymentChannel string

const (
	PaymentChannelAlipay PaymentChannel = "alipay"
)

type PurchasePaymentStatus string

const (
	PurchasePaymentStatusPending  PurchasePaymentStatus = "pending"
	PurchasePaymentStatusPaid     PurchasePaymentStatus = "paid"
	PurchasePaymentStatusExpired  PurchasePaymentStatus = "expired"
	PurchasePaymentStatusRefunded PurchasePaymentStatus = "refunded"
	PurchasePaymentStatusClosed   PurchasePaymentStatus = "closed"
	PurchasePaymentStatusFailed   PurchasePaymentStatus = "failed"
)

type PurchaseFulfillmentStatus string

const (
	PurchaseFulfillmentStatusPending   PurchaseFulfillmentStatus = "pending"
	PurchaseFulfillmentStatusFulfilled PurchaseFulfillmentStatus = "fulfilled"
	PurchaseFulfillmentStatusFailed    PurchaseFulfillmentStatus = "failed"
)

type PurchaseOrderKind string

const (
	PurchaseOrderKindSubscription PurchaseOrderKind = "subscription"
	PurchaseOrderKindBalanceTopup PurchaseOrderKind = "balance_topup"
)

type PurchaseDeliveryMode string

const (
	PurchaseDeliveryModeAccount    PurchaseDeliveryMode = "account"
	PurchaseDeliveryModeRedeemCode PurchaseDeliveryMode = "redeem_code"
)

type AlipayEnvironment string

const (
	AlipayEnvironmentSandbox    AlipayEnvironment = "sandbox"
	AlipayEnvironmentProduction AlipayEnvironment = "production"
)

type PurchaseSettings struct {
	PurchaseEnabled                bool              `json:"purchaseEnabled"`
	DebugAutoPaid                  bool              `json:"debugAutoPaid"`
	AlipayAppID                    string            `json:"alipayAppId"`
	AlipayPID                      string            `json:"alipayPid"`
	AlipayEnvironment              AlipayEnvironment `json:"alipayEnvironment"`
	AlipayNotifyURL                string            `json:"alipayNotifyUrl"`
	AlipayPublicKey                string            `json:"alipayPublicKey"`
	AlipayPrivateKey               string            `json:"-"`
	BalanceTopupPriceCnyCentPerUSD int64             `json:"-"`
}

type PurchaseWebhookEventStatus string

const (
	PurchaseWebhookEventStatusPending    PurchaseWebhookEventStatus = "pending"
	PurchaseWebhookEventStatusProcessing PurchaseWebhookEventStatus = "processing"
	PurchaseWebhookEventStatusSucceeded  PurchaseWebhookEventStatus = "succeeded"
	PurchaseWebhookEventStatusFailed     PurchaseWebhookEventStatus = "failed"
)

type PurchaseSettingsResponse struct {
	PurchaseEnabled                bool              `json:"purchaseEnabled"`
	DebugAutoPaid                  bool              `json:"debugAutoPaid"`
	AlipayAppID                    string            `json:"alipayAppId"`
	AlipayPID                      string            `json:"alipayPid"`
	AlipayEnvironment              AlipayEnvironment `json:"alipayEnvironment"`
	AlipayNotifyURL                string            `json:"alipayNotifyUrl"`
	AlipayPublicKey                string            `json:"alipayPublicKey"`
	PrivateKeySet                  bool              `json:"privateKeySet"`
	PaymentConfigured              bool              `json:"paymentConfigured"`
	BalanceTopupPriceCnyPerUsd     float64           `json:"balanceTopupPriceCnyPerUsd"`
	BalanceTopupPriceCnyCentPerUsd int64             `json:"balanceTopupPriceCnyCentPerUsd"`
}

type PurchaseSettingsRequest struct {
	PurchaseEnabled            bool                     `json:"purchaseEnabled"`
	DebugAutoPaid              bool                     `json:"debugAutoPaid"`
	AlipayAppID                string                   `json:"alipayAppId" binding:"max=64"`
	AlipayPID                  string                   `json:"alipayPid" binding:"max=64"`
	AlipayEnvironment          AlipayEnvironment        `json:"alipayEnvironment" binding:"omitempty,oneof=sandbox production"`
	AlipayNotifyURL            string                   `json:"alipayNotifyUrl" binding:"max=255"`
	AlipayPublicKey            string                   `json:"alipayPublicKey" binding:"max=8192"`
	AlipayPrivateKey           string                   `json:"alipayPrivateKey" binding:"max=8192"`
	BalanceTopupPriceCnyPerUsd *precision.DecimalString `json:"balanceTopupPriceCnyPerUsd,omitempty"`
}

type PurchaseProduct struct {
	ID                 string    `json:"id"`
	Name               string    `json:"name"`
	Summary            string    `json:"summary"`
	SubscriptionPlanID string    `json:"subscriptionPlanId"`
	DurationDays       int       `json:"durationDays"`
	PriceCNYCent       int64     `json:"priceCnyCent"`
	GroupName          string    `json:"groupName"`
	GroupSort          int       `json:"groupSort"`
	IsRecommended      bool      `json:"isRecommended"`
	SortOrder          int       `json:"sortOrder"`
	Enabled            bool      `json:"enabled"`
	CreatedAt          time.Time `json:"createdAt"`
	UpdatedAt          time.Time `json:"updatedAt"`
}

type PurchaseProductRequest struct {
	Name               string `json:"name" binding:"required,min=1,max=64"`
	Summary            string `json:"summary" binding:"max=120"`
	SubscriptionPlanID string `json:"subscriptionPlanId" binding:"required"`
	DurationDays       int    `json:"durationDays" binding:"required,min=1,max=3650"`
	PriceCNYCent       int64  `json:"priceCnyCent" binding:"required,min=1"`
	GroupName          string `json:"groupName" binding:"max=64"`
	GroupSort          int    `json:"groupSort"`
	IsRecommended      bool   `json:"isRecommended"`
	SortOrder          int    `json:"sortOrder"`
	Enabled            bool   `json:"enabled"`
}

type PurchaseProductResponse struct {
	ID                          string    `json:"id"`
	Name                        string    `json:"name"`
	Summary                     string    `json:"summary"`
	SubscriptionPlanID          string    `json:"subscriptionPlanId"`
	SubscriptionPlanName        string    `json:"subscriptionPlanName"`
	SubscriptionPlanUpgradeRank int       `json:"subscriptionPlanUpgradeRank"`
	DurationDays                int       `json:"durationDays"`
	PriceCNYCent                int64     `json:"priceCnyCent"`
	GroupName                   string    `json:"groupName"`
	GroupSort                   int       `json:"groupSort"`
	IsRecommended               bool      `json:"isRecommended"`
	SortOrder                   int       `json:"sortOrder"`
	Enabled                     bool      `json:"enabled"`
	CreatedAt                   time.Time `json:"createdAt"`
	UpdatedAt                   time.Time `json:"updatedAt"`
}

type PurchaseOrder struct {
	ID                      string                    `json:"id"`
	OrderNo                 string                    `json:"orderNo"`
	UserID                  string                    `json:"userId"`
	ProductID               string                    `json:"productId"`
	SubscriptionPlanID      string                    `json:"subscriptionPlanId"`
	DurationDays            int                       `json:"durationDays"`
	AmountCNYCent           int64                     `json:"amountCnyCent"`
	OrderKind               PurchaseOrderKind         `json:"orderKind"`
	DeliveryMode            PurchaseDeliveryMode      `json:"deliveryMode"`
	BalanceTopupMicros      int64                     `json:"balanceTopupMicros"`
	PaymentChannel          PaymentChannel            `json:"paymentChannel"`
	PaymentStatus           PurchasePaymentStatus     `json:"paymentStatus"`
	FulfillmentStatus       PurchaseFulfillmentStatus `json:"fulfillmentStatus"`
	GeneratedRedeemCodeID   string                    `json:"generatedRedeemCodeId"`
	UpgradeSourcePlanID     string                    `json:"upgradeSourcePlanId"`
	UpgradeSourceExpiresAt  *time.Time                `json:"upgradeSourceExpiresAt"`
	UpgradeCreditCNYCent    int64                     `json:"upgradeCreditCnyCent"`
	UpgradeLockedTargetSecs int64                     `json:"upgradeLockedTargetSeconds"`
	UpgradeStateToken       string                    `json:"upgradeStateToken"`
	AlipayTradeNo           string                    `json:"alipayTradeNo"`
	ManualSettlementDone    bool                      `json:"manualSettlementDone"`
	AlipayQRCode            string                    `json:"-"`
	AlipayQRURL             string                    `json:"-"`
	ExpiresAt               *time.Time                `json:"expiresAt"`
	PaidAt                  *time.Time                `json:"paidAt"`
	FulfilledAt             *time.Time                `json:"fulfilledAt"`
	FailureReason           string                    `json:"failureReason"`
	CreatedAt               time.Time                 `json:"createdAt"`
	UpdatedAt               time.Time                 `json:"updatedAt"`
}

type PurchaseOrderResponse struct {
	ID                        string                    `json:"id"`
	OrderNo                   string                    `json:"orderNo"`
	UserID                    string                    `json:"userId"`
	Username                  string                    `json:"username,omitempty"`
	ProductID                 string                    `json:"productId"`
	ProductName               string                    `json:"productName"`
	ProductSummary            string                    `json:"productSummary"`
	SubscriptionPlanID        string                    `json:"subscriptionPlanId"`
	SubscriptionPlanName      string                    `json:"subscriptionPlanName"`
	DurationDays              int                       `json:"durationDays"`
	AmountCNYCent             int64                     `json:"amountCnyCent"`
	OrderKind                 PurchaseOrderKind         `json:"orderKind"`
	DeliveryMode              PurchaseDeliveryMode      `json:"deliveryMode"`
	BalanceTopupMicros        int64                     `json:"balanceTopupMicros"`
	PaymentChannel            PaymentChannel            `json:"paymentChannel"`
	PaymentStatus             PurchasePaymentStatus     `json:"paymentStatus"`
	FulfillmentStatus         PurchaseFulfillmentStatus `json:"fulfillmentStatus"`
	UpgradeSourcePlanID       string                    `json:"upgradeSourcePlanId"`
	UpgradeSourcePlanName     string                    `json:"upgradeSourcePlanName"`
	UpgradeSourceExpiresAt    *time.Time                `json:"upgradeSourceExpiresAt"`
	UpgradeCreditCnyCent      int64                     `json:"upgradeCreditCnyCent"`
	UpgradeLockedTargetSecs   int64                     `json:"upgradeLockedTargetSeconds"`
	GeneratedRedeemCodeID     string                    `json:"generatedRedeemCodeId"`
	GeneratedRedeemCode       string                    `json:"generatedRedeemCode"`
	GeneratedRedeemCodeMask   string                    `json:"generatedRedeemCodeMask"`
	GeneratedRedeemCodeStatus string                    `json:"generatedRedeemCodeStatus"`
	GeneratedRedeemedAt       *time.Time                `json:"generatedRedeemedAt"`
	AlipayTradeNo             string                    `json:"alipayTradeNo"`
	ManualSettlementDone      bool                      `json:"manualSettlementDone"`
	PaymentQRCode             string                    `json:"paymentQrCode"`
	PaymentQRURL              string                    `json:"paymentQrUrl"`
	PaymentQRImageDataURL     string                    `json:"paymentQrImageDataUrl"`
	ExpiresAt                 *time.Time                `json:"expiresAt"`
	PaidAt                    *time.Time                `json:"paidAt"`
	FulfilledAt               *time.Time                `json:"fulfilledAt"`
	FailureReason             string                    `json:"failureReason"`
	CanRefresh                bool                      `json:"canRefresh"`
	CreatedAt                 time.Time                 `json:"createdAt"`
	UpdatedAt                 time.Time                 `json:"updatedAt"`
}

type PurchaseCatalogResponse struct {
	PurchaseEnabled                bool                       `json:"purchaseEnabled"`
	DebugAutoPaid                  bool                       `json:"debugAutoPaid"`
	PaymentConfigured              bool                       `json:"paymentConfigured"`
	RenewalRule                    string                     `json:"renewalRule"`
	CurrentSubscription            *UserSubscriptionResponse  `json:"currentSubscription"`
	Products                       []*PurchaseProductResponse `json:"products"`
	BalanceTopupEnabled            bool                       `json:"balanceTopupEnabled"`
	BalanceTopupPriceCnyPerUsd     float64                    `json:"balanceTopupPriceCnyPerUsd"`
	BalanceTopupPriceCnyCentPerUsd int64                      `json:"balanceTopupPriceCnyCentPerUsd"`
}

type PurchaseOrderListResponse struct {
	Items []*PurchaseOrderResponse `json:"items"`
	Total int64                    `json:"total"`
}

type CreatePurchaseOrderRequest struct {
	ProductID    string               `json:"productId" binding:"required"`
	DeliveryMode PurchaseDeliveryMode `json:"deliveryMode" binding:"omitempty,oneof=account redeem_code"`
}

type CreateBalanceTopupOrderRequest struct {
	AmountUsd precision.DecimalString `json:"amountUsd"`
}

type PurchaseOrderFilters struct {
	PaymentStatus     PurchasePaymentStatus
	FulfillmentStatus PurchaseFulfillmentStatus
	Username          string
	ProductID         string
	Limit             int
	Offset            int
}

type PurchaseWebhookTarget struct {
	ID              string    `json:"id"`
	Name            string    `json:"name"`
	TargetURL       string    `json:"targetUrl"`
	BodyTemplate    string    `json:"bodyTemplate"`
	HeadersTemplate string    `json:"headersTemplate"`
	Enabled         bool      `json:"enabled"`
	CreatedAt       time.Time `json:"createdAt"`
	UpdatedAt       time.Time `json:"updatedAt"`
}

type PurchaseWebhookTargetRequest struct {
	Name            string `json:"name" binding:"required,min=1,max=64"`
	TargetURL       string `json:"targetUrl" binding:"required,max=2048"`
	BodyTemplate    string `json:"bodyTemplate" binding:"max=8192"`
	HeadersTemplate string `json:"headersTemplate" binding:"max=8192"`
	Enabled         bool   `json:"enabled"`
}

type PurchaseWebhookTestResponse struct {
	OK                 bool              `json:"ok"`
	ResponseStatusCode int               `json:"responseStatusCode"`
	ResponseHeaders    map[string]string `json:"responseHeaders"`
	ResponseBody       string            `json:"responseBody"`
	Variables          map[string]string `json:"variables"`
}

type PurchaseWebhookEvent struct {
	ID                 string                     `json:"id"`
	OrderID            string                     `json:"orderId"`
	OrderNo            string                     `json:"orderNo"`
	TargetID           string                     `json:"targetId"`
	Status             PurchaseWebhookEventStatus `json:"status"`
	AttemptCount       int                        `json:"attemptCount"`
	LastError          string                     `json:"lastError"`
	ResponseStatusCode int                        `json:"responseStatusCode"`
	ClaimToken         string                     `json:"claimToken"`
	ClaimedAt          *time.Time                 `json:"claimedAt"`
	ClaimUntil         *time.Time                 `json:"claimUntil"`
	NextAttemptAt      *time.Time                 `json:"nextAttemptAt"`
	LastAttemptAt      *time.Time                 `json:"lastAttemptAt"`
	DeliveredAt        *time.Time                 `json:"deliveredAt"`
	CreatedAt          time.Time                  `json:"createdAt"`
	UpdatedAt          time.Time                  `json:"updatedAt"`
}

type PurchaseOrderPaymentStatusHistory struct {
	ID         string                `json:"id"`
	OrderID    string                `json:"orderId"`
	OrderNo    string                `json:"orderNo"`
	FromStatus PurchasePaymentStatus `json:"fromStatus"`
	ToStatus   PurchasePaymentStatus `json:"toStatus"`
	Note       string                `json:"note"`
	CreatedBy  string                `json:"createdBy"`
	CreatedAt  time.Time             `json:"createdAt"`
}

type PurchaseOrderPaymentStatusUpdateRequest struct {
	PaymentStatus PurchasePaymentStatus `json:"paymentStatus" binding:"required,oneof=paid expired refunded"`
	Note          string                `json:"note" binding:"max=500"`
}

type PurchaseManualSettlementBatchMode string

const (
	PurchaseManualSettlementBatchModeSingle PurchaseManualSettlementBatchMode = "single"
	PurchaseManualSettlementBatchModeBatch  PurchaseManualSettlementBatchMode = "batch"
)

type PurchaseManualSettlementBatch struct {
	ID                 string                            `json:"id"`
	BatchNo            string                            `json:"batchNo"`
	Mode               PurchaseManualSettlementBatchMode `json:"mode"`
	CreatedBy          string                            `json:"createdBy"`
	Note               string                            `json:"note"`
	OrderCount         int                               `json:"orderCount"`
	TotalAmountCNYCent int64                             `json:"totalAmountCnyCent"`
	CreatedAt          time.Time                         `json:"createdAt"`
}

type PurchaseManualSettlementConfirmRequest struct {
	OrderNos []string `json:"orderNos" binding:"required,min=1"`
	Note     string   `json:"note" binding:"max=500"`
}

File: web/src/api/purchase.ts
import type { UserSubscriptionResponse } from './subscription'
import { authFetch } from './client'

const API_BASE = '/api'

async function fetchJson<T>(url: string, options?: RequestInit): Promise<T> {
  const res = await authFetch(url, options)
  let data: unknown

  try {
    data = await res.json()
  } catch {
    data = undefined
  }

  if (!res.ok) {
    const payload = data as { error?: string; message?: string } | undefined
    throw new Error(payload?.error || payload?.message || `请求失败 (${res.status})`)
  }

  return data as T
}

export type PurchasePaymentStatus = 'pending' | 'paid' | 'expired' | 'closed' | 'failed'
export type PurchaseFulfillmentStatus = 'pending' | 'fulfilled' | 'failed'
export type AlipayEnvironment = 'sandbox' | 'production'

export interface PurchaseSettingsResponse {
  purchaseEnabled: boolean
  debugAutoPaid: boolean
  alipayAppId: string
  alipayPid: string
  alipayEnvironment: AlipayEnvironment
  alipayNotifyUrl: string
  alipayPublicKey: string
  privateKeySet: boolean
  paymentConfigured: boolean
  balanceTopupPriceCnyPerUsd: number
}

export interface PurchaseSettingsRequest {
  purchaseEnabled: boolean
  debugAutoPaid: boolean
  alipayAppId: string
  alipayPid: string
  alipayEnvironment: AlipayEnvironment
  alipayNotifyUrl: string
  alipayPublicKey: string
  alipayPrivateKey?: string
  balanceTopupPriceCnyPerUsd?: number
}

export interface PurchaseProduct {
  id: string
  name: string
  summary: string
  subscriptionPlanId: string
  subscriptionPlanName: string
  subscriptionPlanUpgradeRank: number
  durationDays: number
  priceCnyCent: number
  groupName: string
  groupSort: number
  isRecommended: boolean
  sortOrder: number
  enabled: boolean
  createdAt: string
  updatedAt: string
}

export interface PurchaseProductRequest {
  name: string
  summary: string
  subscriptionPlanId: string
  durationDays: number
  priceCnyCent: number
  groupName: string
  groupSort: number
  isRecommended: boolean
  sortOrder: number
  enabled: boolean
}

export interface PurchaseOrder {
  id: string
  orderNo: string
  userId: string
  username?: string
  productId: string
  productName: string
  productSummary: string
  subscriptionPlanId: string
  subscriptionPlanName: string
  durationDays: number
  amountCnyCent: number
  orderKind: 'subscription' | 'balance_topup'
  deliveryMode: 'account' | 'redeem_code'
  balanceTopupMicros: number
  paymentChannel: 'alipay'
  paymentStatus: PurchasePaymentStatus
  fulfillmentStatus: PurchaseFulfillmentStatus
  upgradeSourcePlanId: string
  upgradeSourcePlanName: string
  upgradeSourceExpiresAt: string | null
  upgradeCreditCnyCent: number
  upgradeLockedTargetSeconds: number
  generatedRedeemCodeId: string
  generatedRedeemCode: string
  generatedRedeemCodeMask: string
  generatedRedeemCodeStatus: string
  generatedRedeemedAt: string | null
  alipayTradeNo: string
  paymentQrCode: string
  paymentQrUrl: string
  paymentQrImageDataUrl: string
  expiresAt: string | null
  paidAt: string | null
  fulfilledAt: string | null
  failureReason: string
  canRefresh: boolean
  createdAt: string
  updatedAt: string
}

export interface PurchaseCatalogResponse {
  purchaseEnabled: boolean
  debugAutoPaid: boolean
  paymentConfigured: boolean
  renewalRule: string
  currentSubscription: UserSubscriptionResponse | null
  products: PurchaseProduct[]
  balanceTopupEnabled: boolean
  balanceTopupPriceCnyPerUsd: number
}

export interface PurchaseOrderListResponse {
  items: PurchaseOrder[]
  total: number
}

export async function getPurchaseCatalog(): Promise<PurchaseCatalogResponse> {
  return fetchJson<PurchaseCatalogResponse>(`${API_BASE}/me/purchase/products`)
}

export async function createPurchaseOrder(productId: string): Promise<PurchaseOrder> {
  return fetchJson<PurchaseOrder>(`${API_BASE}/me/purchase/orders`, {
    method: 'POST',
    body: JSON.stringify({ productId, deliveryMode: 'account' }),
  })
}

export async function createPurchaseOrderWithMode(productId: string, deliveryMode: 'account' | 'redeem_code'): Promise<PurchaseOrder> {
  return fetchJson<PurchaseOrder>(`${API_BASE}/me/purchase/orders`, {
    method: 'POST',
    body: JSON.stringify({ productId, deliveryMode }),
  })
}

export async function createBalanceTopupOrder(amountUsd: number): Promise<PurchaseOrder> {
  return fetchJson<PurchaseOrder>(`${API_BASE}/me/purchase/balance-topup/orders`, {
    method: 'POST',
    body: JSON.stringify({ amountUsd }),
  })
}

export async function listMyPurchaseOrders(): Promise<PurchaseOrderListResponse> {
  return fetchJson<PurchaseOrderListResponse>(`${API_BASE}/me/purchase/orders`)
}

export async function getMyPurchaseOrder(orderNo: string): Promise<PurchaseOrder> {
  return fetchJson<PurchaseOrder>(`${API_BASE}/me/purchase/orders/${encodeURIComponent(orderNo)}`)
}

export async function refreshMyPurchaseOrder(orderNo: string): Promise<PurchaseOrder> {
  return fetchJson<PurchaseOrder>(`${API_BASE}/me/purchase/orders/${encodeURIComponent(orderNo)}/refresh`, {
    method: 'POST',
  })
}

export async function getPurchaseSettings(): Promise<PurchaseSettingsResponse> {
  return fetchJson<PurchaseSettingsResponse>(`${API_BASE}/admin/purchase/settings`)
}

export async function updatePurchaseSettings(payload: PurchaseSettingsRequest): Promise<{ message: string; settings: PurchaseSettingsResponse }> {
  return fetchJson<{ message: string; settings: PurchaseSettingsResponse }>(`${API_BASE}/admin/purchase/settings`, {
    method: 'PUT',
    body: JSON.stringify(payload),
  })
}

export async function listPurchaseProductsAdmin(): Promise<PurchaseProduct[]> {
  const data = await fetchJson<{ products: PurchaseProduct[] }>(`${API_BASE}/admin/purchase/products`)
  return data.products || []
}

export async function createPurchaseProduct(payload: PurchaseProductRequest): Promise<PurchaseProduct> {
  return fetchJson<PurchaseProduct>(`${API_BASE}/admin/purchase/products`, {
    method: 'POST',
    body: JSON.stringify(payload),
  })
}

export async function updatePurchaseProduct(id: string, payload: PurchaseProductRequest): Promise<PurchaseProduct> {
  return fetchJson<PurchaseProduct>(`${API_BASE}/admin/purchase/products/${encodeURIComponent(id)}`, {
    method: 'PUT',
    body: JSON.stringify(payload),
  })
}

export async function deletePurchaseProduct(id: string): Promise<void> {
  await fetchJson<{ message: string }>(`${API_BASE}/admin/purchase/products/${encodeURIComponent(id)}`, {
    method: 'DELETE',
  })
}

export async function setPurchaseProductEnabled(id: string, enabled: boolean): Promise<void> {
  await fetchJson<{ message: string }>(`${API_BASE}/admin/purchase/products/${encodeURIComponent(id)}/enabled`, {
    method: 'PATCH',
    body: JSON.stringify({ enabled }),
  })
}

export interface AdminOrderFilters {
  paymentStatus?: PurchasePaymentStatus | ''
  fulfillmentStatus?: PurchaseFulfillmentStatus | ''
  username?: string
  productId?: string
  limit?: number
}

export async function listPurchaseOrdersAdmin(filters: AdminOrderFilters = {}): Promise<PurchaseOrderListResponse> {
  const params = new URLSearchParams()
  if (filters.paymentStatus) params.set('paymentStatus', filters.paymentStatus)
  if (filters.fulfillmentStatus) params.set('fulfillmentStatus', filters.fulfillmentStatus)
  if (filters.username) params.set('username', filters.username)
  if (filters.productId) params.set('productId', filters.productId)
  if (filters.limit) params.set('limit', String(filters.limit))
  const query = params.toString()
  return fetchJson<PurchaseOrderListResponse>(`${API_BASE}/admin/purchase/orders${query ? `?${query}` : ''}`)
}

export async function refreshPurchaseOrderAdmin(orderNo: string): Promise<PurchaseOrder> {
  return fetchJson<PurchaseOrder>(`${API_BASE}/admin/purchase/orders/${encodeURIComponent(orderNo)}/refresh`, {
    method: 'POST',
  })
}
